### PR TITLE
Telemetry Operator: Input documentation

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**

Changes proposed in this pull request:

- Add documentation on creating the input section

**Related issue(s)**
* https://github.com/kyma-project/kyma/issues/14564

### CODE REVIEW

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 1]
```diff
@@ -52,38 +52,88 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+        application:
+          namespaces: []
+          excludeNamespaces: []
+          containers: []
+          excludeContainers: []
+          includeSystemNamespaces: false
+    ```
+
+Selectors:
+- `namespaces`: a list of namespaces logs should be collected from (mutually exclusive with `exludeNamespaces`)
```
Comment (chrkl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:89]: Maybe we move this to the "Parameters" section at the end?

Comment (lindnerby) [reply docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md]: Of course... done 👍

**Review by chrkl — APPROVED** [review]

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 2]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
-          value: "user"
-        password:
-          value: "not-required"
-        tls:
-          disabled: true
-          skipCertificateValidation: true
-   ```
+## Setting up a LogPipeline
+
+### Step 1: Create an Input
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:57]:
```suggestion
### Step 1: Create an input
```
Our guidelines want [sentence case](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology/#capitalization) in headlines.

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 3]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+  ```yaml
+  spec:
+    input:
+      application:
+        includeSystemNamespaces: true
+        excludeContainers:
+          - fluent-bit
+  ```
+
+### Step 2: Create an Output
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:98]:
```suggestion
### Step 2: Create an output
```
Our guidelines want [sentence case](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology/#capitalization) in headlines.

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 4]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
-          value: "not-required"
-        tls:
-          disabled: true
-          skipCertificateValidation: true
-   ```
+## Setting up a LogPipeline
+
+### Step 1: Create an Input
+
+1. To create a LogPipeline and define from which resources logs should be included create a LogPipeline with a name and spec with input:
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:59]: Trying to shorten and simplify this:

```suggestion
1. Create a LogPipeline with a `name` and `spec` with **input**, which defines from which resources logs are included:
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 5]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+    ```yaml
+    kind: LogPipeline
+    apiVersion: telemetry.kyma-project.io/v1alpha1
+    metadata:
+      name: http-backend
+    spec:
+      input: ...
+    ```
+
+2. As of now only `application` input can be set, meaning the LogPipeline will collect application logs, which is also the default, when omitting the input altogether.
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:69]: We should [avoid words like "currently"](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology/#terminology) etc. 

Btw, if it's the default and the only option at the moment, why do we instruct the user to define it at all? Is this step necessary as long as we only support application logs? How about phrasing it as a "Note" as long as there's no other option? We can resurrect it as a step once more kinds of logs are supported.

```suggestion
   > **NOTE:** By default, the LogPipeline collects application logs, so you don't have to specify the input. To specify the input, run:
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 6]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+    ```yaml
+    spec:
+      input:
+        application: ...
+    ```
+
+3. Selection mechanisms for application logs can be applied on namespace or container level. If nothing is set all namespaces will be considered except the system namespaces `kube-system`, `istio-system`, `kyma-system`, `kyma-integration` which are excluded by default.
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:76]: We prefer [active voice](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#style-and-terminology-active-voice) :)

```suggestion
3. Optionally, you can apply selection mechanisms for application logs on Namespace or container level.
   If nothing is set, all Namespaces are considered, except the system Namespaces `kube-system`, `istio-system`, `kyma-system`, and `kyma-integration`, which are excluded by default.
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 7]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+      input:
+        application:
+          namespaces: []
+          excludeNamespaces: []
+          containers: []
+          excludeContainers: []
+          includeSystemNamespaces: false
+    ```
+
+Valid input example:
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:88]: If this belongs to step 3, add 3 blank spaces indentation in this and the following lines.
If this is separate of step 3, what is it? A summary of the instructions? Then we could define it as a result of the whole "Step 1: Create an input" instructions.

```suggestion
### Result

You've created a valid input. See the following example:
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 8]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+    input:
+      application:
+        includeSystemNamespaces: true
+        excludeContainers:
+          - fluent-bit
+  ```
+
+### Step 2: Create an Output
+
+1. To ship application logs to a new output, define an output spec for the LogPipeline like this:
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:100]:
```suggestion
1. To ship application logs to a new output, define an output spec for the LogPipeline:
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 9]
```diff
@@ -52,38 +52,81 @@ The LogPipeline resource is managed by the Telemetry Operator, a typical Kuberne
...
+          password:
+            value: "not-required"
+          tls:
+            disabled: true
+            skipCertificateValidation: true
+      input:
+        ...
+    ```
 
    An output is a data destination configured by a [Fluent Bit output](https://docs.fluentbit.io/manual/pipeline/outputs) of the relevant type. The LogPipeline supports the following output types:
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:131]: Consider putting this explanatory paragraph (lines 131-134) below the headline (line 98), above the first step (line 100).

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 10]
```diff
@@ -125,7 +163,7 @@ The Telemetry Operator watches all LogPipeline resources and related Secrets. Wh
    http-backend      Ready     44s
    ```
 
-### Step 2: Add filters
+### Step 3: Add filters
 
 To enrich logs with attributes or drop whole lines, add filters to the existing pipeline. In the following example, three filters are added, which are executed in sequence.
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:168]: Might be simpler this way:

```suggestion
To enrich logs with attributes or drop whole lines, add filters to the existing pipeline. 
The following example contains three filters, which are executed in sequence.
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 11]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
 
 | Parameter | Type | Description |
 |---|---|---|
+| input | object | Definition where to collect logs including selector mechanims. |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:314]:
```suggestion
| input | object | Definition where to collect logs, including selector mechanisms. |
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 12]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
+| input.application | object | Input type for application logs collection. |
+| input.application.namespaces | []string | List of namespaces logs should be collected from (mutually exclusive with `exludeNamespaces`). |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:316]:
```suggestion
| input.application.namespaces | []string | List of Namespaces from which logs are collected (mutually exclusive with `exludeNamespaces`). |
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 13]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
+| input.application.excludeNamespaces | []string | List namespaces to exclude from log collection from all namespaces (mutually exclusive with `namespaces`) |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:317]:
```suggestion
| input.application.excludeNamespaces | []string | List of Namespaces to exclude during log collection from all Namespaces (mutually exclusive with `namespaces`) |
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 14]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
+| input.application.containers | []string | List of container names to collect from (mutually exclusive with `excludeContainers`) |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:318]:
```suggestion
| input.application.containers | []string | List of containers to collect from (mutually exclusive with `excludeContainers`) |
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 15]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
+| input.application.excludeContainers | []string | List of container names to exclude (mutually exclusive with `containers`) |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:319]:
```suggestion
| input.application.excludeContainers | []string | List of containers to exclude (mutually exclusive with `containers`) |
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 16]
```diff
@@ -267,6 +311,13 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
+| input.application.includeSystemNamespaces | boolean | If no selection from `namespaces` or `exludeNamespaces` is done, this can be set to include all system namespaces without listing them manually. Defaults to `false`. |
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:320]:
```suggestion
| input.application.includeSystemNamespaces | boolean | If you specify neither `namespaces` nor `excludeNamespaces`, you can set `includeSystemNamespaces` to include all system Namespaces without listing them manually. Defaults to `false`. |
```

**Review by chrkl — APPROVED** [review]

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 17]
```diff
@@ -124,48 +82,75 @@ Valid input example:
...
 3. Check that the status of the LogPipeline in your cluster is `Ready`:
+    ```bash
+    kubectl get logpipeline
+    NAME              STATUS    AGE
+    http-backend      Ready     44s
+    ```
+
+### Step 2: Create an input
+
+To restrict or specify from which resources logs should be included an input spec can be defined optionally, if selection mechanisms for application logs on Namespace or container level are needed.
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:120]: Passive => Active voice

```suggestion
If you need selection mechanisms for application logs on Namespace or container level, you can restrict or specify from which resources logs are included with an input spec.
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 18]
```diff
@@ -124,48 +82,75 @@ Valid input example:
...
+Defining an input can be omitted entireley to collect from all Namespaces, except the system Namespaces `kube-system`, `istio-system`, `kyma-system`, and `kyma-integration`, which are excluded by default.
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:121]:
```suggestion
If you don't define any input, it's collected from all Namespaces, except the system Namespaces `kube-system`, `istio-system`, `kyma-system`, and `kyma-integration`, which are excluded by default. 
```

File: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md [hunk 19]
```diff
@@ -124,48 +82,75 @@ Valid input example:
...
+  output:
+    ...
+```
 
-   ```bash
-   kubectl get logpipeline
-   NAME              STATUS    AGE
-   http-backend      Ready     44s
-   ```
+Example, to collect from all namespaces including system namespaces but excluding on container level from fluent-bit:
```
Comment (NHingerl) [inline docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md:140]:
```suggestion
The following example collects input from all Namespaces including system Namespaces, but excludes the Fluent Bit container:
```

**Review by NHingerl — APPROVED** [review]

**Pull request merged**